sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs

```diff
@@ -56,55 +42,48 @@ public class AnalyzeHealthcareEntitiesOperation : Operation<AnalyzeHealthcareEnt
         public TextAnalyticsOperationStatus Status => _status;
 
         /// <summary>
-        /// next link string for pagination
-        /// </summary>
-        internal string NextLink { get; set; }
-
-        /// <summary>
-        /// Final result of the long-running operation.
+        /// Gets the final result of the long-running operation in a synchronous way.
         /// </summary>
         /// <remarks>
         /// This property can be accessed only after the operation completes successfully (HasValue is true).
         /// </remarks>
-        public override AnalyzeHealthcareEntitiesResultCollection Value
-        {
-            get
-            {
-#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
-                if (HasCompleted && !HasValue)
-                    throw _requestFailedException;
-#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
-                else
-                {
-                    return OperationHelpers.GetValue(ref _value);
-                }
-            }
-        }
-
-        /// <summary><c>true</c> if the long-running operation has completed. Otherwise, <c>false</c>.</summary>
-        private bool _hasCompleted;
+        public override AsyncPageable<AnalyzeHealthcareEntitiesResultCollection> Value => GetValuesAsync();
 
         /// <summary>
         /// Returns true if the long-running operation completed.
         /// </summary>
         public override bool HasCompleted => _hasCompleted;
 
+        /// <summary>
+        /// Returns true if the long-running operation completed successfully and has produced final result (accessible by Value property).
+        /// </summary>
+        public override bool HasValue => _firstPage != null;
+
+        /// <summary>Provides communication with the Text Analytics Azure Cognitive Service through its REST API.</summary>
+        private readonly TextAnalyticsRestClient _serviceClient;
+
+        /// <summary>Provides tools for exception creation in case of failure.</summary>
+        private readonly ClientDiagnostics _diagnostics;
+
+        /// <summary><c>true</c> if the long-running operation has completed. Otherwise, <c>false</c>.</summary>
+        private bool _hasCompleted;
+
+        private TextAnalyticsOperationStatus _status;
```

Comment:
I like all the comments - should we add them to the other private fields?

Reply:
added